src/mscorlib/src/System/String.cs

```diff
@@ -973,17 +973,20 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             if ((count == 0) || (omitEmptyEntries && this.Length == 0)) 
             {           
-                return new String[0];
+                return EmptyArray<String>.Value;
```

Comment:
In theory I think this change to use a cached empty array is good.  In practice I do worry about compat issues that could arise.  If it were just for .NET Core, I'd be less concerned, and we've made many similar changes in corefx, but as this mscorlib is used elsewhere, I wonder if it's worth it, especially considering what I'd expect to be the relative rarity of hitting these cases (I could be wrong).  @jkotas, any thoughts?